File: java/strategy/strategy/src/main/java/edu/uea/Bill.java
package edu.uea;

// Contexto
class Bill {
    private DiscountStrategy discountStrategy;

    public Bill(DiscountStrategy discountStrategy) {
        this.discountStrategy = discountStrategy;
    }

    public void setDiscountStrategy(DiscountStrategy discountStrategy) {
        this.discountStrategy = discountStrategy;
    }

    public double calculateFinalAmount(double amount) {
        return discountStrategy.applyDiscount(amount);
    }
}

File: python/strategy/Bill.py
from DiscountStrategy import DiscountStrategy


# Contexto
class Bill:
    def __init__(self, discount_strategy: DiscountStrategy):
        self.discount_strategy = discount_strategy

    def set_discount_strategy(self, discount_strategy: DiscountStrategy):
        self.discount_strategy = discount_strategy

    def calculate_final_amount(self, amount):
        return self.discount_strategy.apply_discount(amount)
